test/dynamics/solvers/test_solver_classes.py

```diff
@@ -1182,6 +1182,63 @@ def setUp(self):
         super().setUp()
         self.method = "jax_odeint"
 
+    def test_t_eval_t_span_jax_odeint(self):
+        """Test internal jitting works when specifying t_eval and t_span. This catches a bug
```

Comment:
Why are these done through pulse schedules and not raw signals?

Reply:
The purpose of this PR and this test are explicitly to fix jitting in the pulse schedule simulation context.